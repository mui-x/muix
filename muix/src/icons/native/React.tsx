import createSvgIcon from './lib/create-svg-icon'

export default createSvgIcon(
  'react',
  'React',
  true
)