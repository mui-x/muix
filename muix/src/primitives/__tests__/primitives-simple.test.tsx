import React from 'react'

import { initPlatform, render } from "reactxx-primitives-test/init-platform.t"
import { platform } from 'reactxx-styles'

describe("PRIMITIVES SIMPLE", () => {

  const doTest = (isWeb: boolean) => {
    beforeEach(() => {
      initPlatform(isWeb, { dataTraceFlag: 'short' })
    })
    it('01: Text', () => {
      const wrapper = render(<platform.Text>Text</platform.Text>)
      expect(wrapper.container).toMatchSnapshot()
      wrapper.unmount()
    })
    it('02: Text singleLine', () => {
      const wrapper = render(<platform.Text singleLine>Text singleLine</platform.Text>)
      expect(wrapper.container).toMatchSnapshot()
      wrapper.unmount()
    })
    it('03: AnimatedText', () => {
      const wrapper = render(<platform.AnimatedText>AnimatedText</platform.AnimatedText>)
      expect(wrapper.container).toMatchSnapshot()
      wrapper.unmount()
    })
    it('04: View', () => {
      const wrapper = render(<platform.View>View</platform.View>)
      expect(wrapper.container).toMatchSnapshot()
      wrapper.unmount()
    })
    it('05: AnimatedView', () => {
      const wrapper = render(<platform.AnimatedView>AnimatedView</platform.AnimatedView>)
      expect(wrapper.container).toMatchSnapshot()
      wrapper.unmount()
    })
    it('06: Icon', () => {
      const wrapper = render(<platform.Icon>Icon</platform.Icon>)
      expect(wrapper.container).toMatchSnapshot()
      wrapper.unmount()
    })
    it('07: AnimatedIcon', () => {
      const wrapper = render(<platform.AnimatedIcon>AnimatedIcon</platform.AnimatedIcon>)
      expect(wrapper.container).toMatchSnapshot()
      wrapper.unmount()
    })
    it('08: ScrollView', () => {
      const wrapper = render(<platform.ScrollView>ScrollView</platform.ScrollView>)
      expect(wrapper.container).toMatchSnapshot()
      wrapper.unmount()
    })
    it('09: ScrollView horizontal', () => {
      const wrapper = render(<platform.ScrollView horizontal>ScrollView</platform.ScrollView>)
      expect(wrapper.container).toMatchSnapshot()
      wrapper.unmount()
    })
  }

  describe("## NATIVE ##", () => doTest(false))
  describe("## WEB ##", () => doTest(true))

})