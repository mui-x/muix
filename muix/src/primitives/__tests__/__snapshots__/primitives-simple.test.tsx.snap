// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PRIMITIVES SIMPLE ## NATIVE ## 01: Text 1`] = `
<div>
  <div
    data-type="TEXT"
  >
    Text
  </div>
</div>
`;

exports[`PRIMITIVES SIMPLE ## NATIVE ## 02: Text singleLine 1`] = `
<div>
  <div
    data-numberoflines="1"
    data-type="TEXT"
  >
    Text singleLine
  </div>
</div>
`;

exports[`PRIMITIVES SIMPLE ## NATIVE ## 03: AnimatedText 1`] = `
<div>
  <div
    data-type="ANIMATEDTEXT"
  >
    AnimatedText
  </div>
</div>
`;

exports[`PRIMITIVES SIMPLE ## NATIVE ## 04: View 1`] = `
<div>
  <div
    data-type="TOUCHABLEWITHOUTFEEDBACK"
  >
    <div
      data-type="VIEW"
    >
      View
    </div>
  </div>
</div>
`;

exports[`PRIMITIVES SIMPLE ## NATIVE ## 05: AnimatedView 1`] = `
<div>
  <div
    data-type="TOUCHABLEWITHOUTFEEDBACK"
  >
    <div
      data-type="ANIMATEDVIEW"
    >
      AnimatedView
    </div>
  </div>
</div>
`;

exports[`PRIMITIVES SIMPLE ## NATIVE ## 06: Icon 1`] = `
<div>
  <div
    data-name="Icon"
    data-type="MATERIALCOMMUNITYICONS"
  />
</div>
`;

exports[`PRIMITIVES SIMPLE ## NATIVE ## 07: AnimatedIcon 1`] = `
<div>
  <div
    data-name="AnimatedIcon"
    data-type="ANIMATED"
  />
</div>
`;

exports[`PRIMITIVES SIMPLE ## NATIVE ## 08: ScrollView 1`] = `
<div>
  <div
    data-type="SCROLLVIEW"
  >
    ScrollView
  </div>
</div>
`;

exports[`PRIMITIVES SIMPLE ## NATIVE ## 09: ScrollView horizontal 1`] = `
<div>
  <div
    data-type="SCROLLVIEW"
  >
    ScrollView
  </div>
</div>
`;

exports[`PRIMITIVES SIMPLE ## WEB ## 01: Text 1`] = `
<div>
  <div
    class="d a c b reactxx-text"
    data-trace="
.d { cursor:pointer }
.a .reactxx-text { display:inline }
.c { word-wrap:break-word }
.b { white-space:pre-wrap }"
  >
    Text
  </div>
</div>
`;

exports[`PRIMITIVES SIMPLE ## WEB ## 02: Text singleLine 1`] = `
<div>
  <div
    class="i h g f e d a c reactxx-text"
    data-trace="
.i { white-space:nowrap }
.h { text-overflow:ellipsis }
.g { overflow:hidden }
.f { max-width:100% }
.e { flex-shrink:1 }
.d { cursor:pointer }
.a .reactxx-text { display:inline }
.c { word-wrap:break-word }"
  >
    Text singleLine
  </div>
</div>
`;

exports[`PRIMITIVES SIMPLE ## WEB ## 03: AnimatedText 1`] = `
<div>
  <div
    class="d a c b reactxx-text"
    data-trace="
.d { cursor:pointer }
.a .reactxx-text { display:inline }
.c { word-wrap:break-word }
.b { white-space:pre-wrap }"
  >
    AnimatedText
  </div>
</div>
`;

exports[`PRIMITIVES SIMPLE ## WEB ## 04: View 1`] = `
<div>
  <div
    class="h g f e d c b a"
    data-trace="
.h { cursor:pointer }
.g { overflow:hidden }
.f { position:relative }
.e { flex-shrink:0 }
.d { flex-basis:auto }
.c { align-items:stretch;-webkit-box-align:stretch }
.b { flex-direction:column;-webkit-box-orient:vertical;-webkit-box-direction:normal }
.a { display:-webkit-box;display:-moz-box;display:-ms-flexbox;display:-webkit-flex;display:flex }"
  >
    View
  </div>
</div>
`;

exports[`PRIMITIVES SIMPLE ## WEB ## 05: AnimatedView 1`] = `
<div>
  <div
    class="h g f e d c b a"
    data-trace="
.h { cursor:pointer }
.g { overflow:hidden }
.f { position:relative }
.e { flex-shrink:0 }
.d { flex-basis:auto }
.c { align-items:stretch;-webkit-box-align:stretch }
.b { flex-direction:column;-webkit-box-orient:vertical;-webkit-box-direction:normal }
.a { display:-webkit-box;display:-moz-box;display:-ms-flexbox;display:-webkit-flex;display:flex }"
  >
    AnimatedView
  </div>
</div>
`;

exports[`PRIMITIVES SIMPLE ## WEB ## 06: Icon 1`] = `
<div>
  <svg
    class="d c b a"
    data-trace="
.d { cursor:pointer }
.c { font-size:inherited }
.b { fill:currentColor }
.a { flex-shrink:0 }"
    focusable="false"
    viewBox="0 0 24 24"
  >
    Icon
  </svg>
</div>
`;

exports[`PRIMITIVES SIMPLE ## WEB ## 07: AnimatedIcon 1`] = `
<div>
  <svg
    class="d c b a"
    data-trace="
.d { cursor:pointer }
.c { font-size:inherited }
.b { fill:currentColor }
.a { flex-shrink:0 }"
    focusable="false"
    viewBox="0 0 24 24"
  >
    AnimatedIcon
  </svg>
</div>
`;

exports[`PRIMITIVES SIMPLE ## WEB ## 08: ScrollView 1`] = `
<div>
  <div
    class="m l k j i h g f e d c b"
    data-trace="
.m { transform:translateZ(0) }
.l { -webkit-overflow-scrolling:touch }
.k { overflow-y:auto }
.j { overflow-x:hidden }
.i { flex-grow:1;-webkit-box-flex:1 }
.h { overflow:hidden }
.g { position:relative }
.f { flex-shrink:0 }
.e { flex-basis:auto }
.d { align-items:stretch;-webkit-box-align:stretch }
.c { flex-direction:column;-webkit-box-orient:vertical;-webkit-box-direction:normal }
.b { display:-webkit-box;display:-moz-box;display:-ms-flexbox;display:-webkit-flex;display:flex }"
  >
    <div
      class="h g f e d c b"
      data-trace="
.h { overflow:hidden }
.g { position:relative }
.f { flex-shrink:0 }
.e { flex-basis:auto }
.d { align-items:stretch;-webkit-box-align:stretch }
.c { flex-direction:column;-webkit-box-orient:vertical;-webkit-box-direction:normal }
.b { display:-webkit-box;display:-moz-box;display:-ms-flexbox;display:-webkit-flex;display:flex }"
    >
      ScrollView
    </div>
  </div>
</div>
`;

exports[`PRIMITIVES SIMPLE ## WEB ## 09: ScrollView horizontal 1`] = `
<div>
  <div
    class="p o n m l i h g f e d b"
    data-trace="
.p { overflow-y:hidden }
.o { overflow-x:auto }
.n { flex-direction:row;-webkit-box-orient:horizontal;-webkit-box-direction:normal }
.m { transform:translateZ(0) }
.l { -webkit-overflow-scrolling:touch }
.i { flex-grow:1;-webkit-box-flex:1 }
.h { overflow:hidden }
.g { position:relative }
.f { flex-shrink:0 }
.e { flex-basis:auto }
.d { align-items:stretch;-webkit-box-align:stretch }
.b { display:-webkit-box;display:-moz-box;display:-ms-flexbox;display:-webkit-flex;display:flex }"
  >
    <div
      class="n h g f e d b"
      data-trace="
.n { flex-direction:row;-webkit-box-orient:horizontal;-webkit-box-direction:normal }
.h { overflow:hidden }
.g { position:relative }
.f { flex-shrink:0 }
.e { flex-basis:auto }
.d { align-items:stretch;-webkit-box-align:stretch }
.b { display:-webkit-box;display:-moz-box;display:-ms-flexbox;display:-webkit-flex;display:flex }"
    >
      ScrollView
    </div>
  </div>
</div>
`;
